core/src/value/deserialize.rs

```diff
@@ -8,6 +8,26 @@ use std::ops::{Add, Mul, Sub};
 pub enum Error {
     #[snafu(display("Unexpected end of element"))]
     UnexpectedEndOfElement { backtrace: Backtrace },
+    //TODO impl Display for DateComponent and Range or leave it like this ?
+    #[snafu(display(
+        "{:?} has invalid value:  {}. Must be in {:?}",
```

Comment:
Sticking to the use of Debug here is fine, since they already print useful and lean textual representations. Leaving just a minor suggestion for the error message. 
```suggestion
        "Invalid {:?} value {}, must be in {:?}",
```